modules/gateways/blockonomics/upgrade.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * This script will delete old files from previuos 1.8.2 version.
+ * First delete files and folder that we know belong to blockonomics only
+ * Then we delete the folders if they are empty
+ * We also check if the screeshots folder and README.md file was installed and belongs to blockonomics so we can delete them too.
+ */
+
+require_once __DIR__ . '/../../../init.php';
+require_once __DIR__ . '/blockonomics.php';
+
+use Blockonomics\Blockonomics;
+
+$blockonomics = new Blockonomics();
+if (doubleval($blockonomics->getVersion()) < 1.9) {
```

Comment:
The upgrade can run automatically from the settings page, similar to when creating the DB tables. This can further simplify upgrades and allow all version-based upgrade code to be added to this file in the future.

Reply:
I think it is better to allow the user to be aware that an upgrade is taking place. Both WHMCS and other famous CMS's like Wordpress, Prestashopo, Magento, etc work that way.  They warn you that there is a new version, but they do not update it without the manual intervention of the administrator.